src/langevin/utils.py

```
@@ -471,3 +476,234 @@ def igarch_volatility_scale(
+    test_sharpe = _evaluate(best_point, test_prices, test_returns, seed + len(grid_points))
+
+    return {
+        'best_params': best_params_full,
+        'best_train_sharpe': best_train_sharpe,
+        'test_sharpe': test_sharpe,
+        'train_test_gap': best_train_sharpe - (test_sharpe if np.isfinite(test_sharpe) else 0.0),
+        'all_results': all_results,
+        'n_evaluated': len(grid_points),
+    }
+
+    return positions
```

Ruff (0.15.6)

[error] Undefined name `positions`

(F821)